Sources/Sentry/SentryTransactionContext.mm

```diff
@@ -74,14 +91,30 @@ - (instancetype)initWithName:(NSString *)name
                                spanId:spanId
                              parentId:parentSpanId
                             operation:operation
-                              sampled:false]) {
+                              sampled:kSentryDefaultSamplingDecision]) {
         _name = [NSString stringWithString:name];
         _nameSource = source;
         self.parentSampled = parentSampled;
+        [self getThreadInfo];
     }
     return self;
 }
 
+- (void)getThreadInfo
```

Comment:
`m`: It would be great to add some tests for this new functionality.

Reply:
We've got the `ThreadHandle` system under test [here](https://github.com/getsentry/sentry-cocoa/blob/master/Tests/SentryTests/Profiling/SentryThreadMetadataCacheTests.mm) and [here](https://github.com/getsentry/sentry-cocoa/blob/master/Tests/SentryTests/Profiling/SentryThreadHandleTests.mm), and `SentryThread` has unit tests [here](https://github.com/getsentry/sentry-cocoa/blob/master/Tests/SentryTests/SentryInterfacesTests.m#L255). 

We test that this field is nonempty [here](https://github.com/getsentry/sentry-cocoa/pull/2358/files#diff-b341a165bb5e23dbf87f2cbbcd39bdf0dbef45c3e57a46e8cda4d49bd4019a0eL331) but I don't think we should use this spot as another place to test whether we're getting what we think is the correct thread info, it's not scalable to do this for every usage. If we think there's an issue with `ThreadHandle` or `SentryThread` let's take care of it separately.